packages/mdc-fab/_mixins.scss

```diff
@@ -21,124 +21,220 @@
 @import "@material/animation/functions";
 @import "@material/elevation/mixins";
 @import "@material/elevation/variables";
+@import "@material/feature-targeting/functions";
+@import "@material/feature-targeting/mixins";
 @import "@material/ripple/mixins";
 @import "@material/ripple/variables";
 @import "@material/theme/functions";
 @import "@material/theme/mixins";
 @import "@material/shape/mixins";
 @import "@material/shape/functions";
 @import "@material/rtl/mixins";
+@import "@material/typography/mixins";
 @import "./variables";
 
-@mixin mdc-fab-accessible($container-color) {
+@mixin mdc-fab-core-styles($query: mdc-feature-all()) {
+  @include mdc-fab-without-ripple($query);
+  @include mdc-fab-ripple($query);
+}
+
+@mixin mdc-fab-without-ripple($query: mdc-feature-all()) {
+  // postcss-bem-linter: define fab
+
+  .mdc-fab {
+    @include mdc-fab-base_($query: $query);
+    @include mdc-fab-container-color(secondary, $query: $query);
+    @include mdc-fab-icon-size(24px, $query: $query);
+    @include mdc-fab-ink-color(on-secondary, $query: $query);
+  }
+
+  .mdc-fab--mini {
+    @include mdc-fab--mini_($query: $query);
+  }
+
+  .mdc-fab--extended {
+    @include mdc-fab--extended_($query: $query);
+  }
+
+  .mdc-fab__label {
+    @include mdc-fab--label_($query: $query);
+  }
+
+  .mdc-fab__icon {
+    @include mdc-fab__icon_($query: $query);
+  }
+
+  // Increase specificity for FAB icon styles that need to override styles defined for .material-icons
+  // (which is loaded separately so the order of CSS definitions is not guaranteed)
+  .mdc-fab .mdc-fab__icon {
+    @include mdc-fab__icon-overrides_($query: $query);
+  }
+
+  .mdc-fab--exited {
+    @include mdc-fab--exited_($query: $query);
+  }
+
+  // postcss-bem-linter: end
+}
+
+@mixin mdc-fab-ripple($query: mdc-feature-all()) {
+  @include mdc-ripple-common($query);
+
+  // postcss-bem-linter: define fab
+
+  .mdc-fab {
+    @include mdc-ripple-surface($query: $query);
+    @include mdc-ripple-radius-bounded($query: $query);
+    @include mdc-states(on-secondary, $query: $query);
+  }
+
+  // postcss-bem-linter: end
+}
+
+@mixin mdc-fab-accessible($container-color, $query: mdc-feature-all()) {
+  $feat-color: mdc-feature-create-target($query, color);
   $fill-tone: mdc-theme-tone($container-color);
 
-  @include mdc-fab-container-color($container-color);
+  @include mdc-fab-container-color($container-color, $query: $query);
 
-  @if ($fill-tone == "dark") {
-    @include mdc-fab-ink-color(text-primary-on-dark);
-    @include mdc-states(text-primary-on-dark);
-  } @else {
-    @include mdc-fab-ink-color(text-primary-on-light);
-    @include mdc-states(text-primary-on-light);
+  @include mdc-feature-targets($feat-color) {
```

Comment:
I think you can remove this one since you're passing the query through below

Reply:
Good point... I had originally forgotten to forward the query through the includes below, and then forgot about the include here.

...Meanwhile, now I'm spooked because this somehow didn't fail building? Seems like it should have...

Reply:
Yeah that is strange, I can see how the feature-targeting test would miss it. We gave it a query that selects nothing, so it fails the first feature-targets check and Sass doesn't bother evaluating what's inside.

But is this getting built as part of the dev app? https://github.com/material-components/material-components-web/blob/f310e7364f6f7a7ae162908ca27808f0e4619184/demos/fab.scss#L93 I would have expected that to result in an error

Reply:
Oh actually it did fail when I checked out your PR, reverted the fix commit and tried to `npm run dev`. Got the following error:

```
ERROR in ./demos/fab.scss
    Module build failed: 
        @error "mdc-feature-targets must not be used inside of another mdc-feature-targets block";
```

I guess maybe this isn't built as part of your CI?

Reply:
I made a PR (https://github.com/material-components/material-components-web/pull/4528) to run all the test mixins twice, once with none of the features and once with all of them. In the all of them case, we don't need to do anything with the output, just make sure that it compiles. I think this would have caught your issue here